app/webpacker/components/ScrambleMatcher/index.jsx

```diff
@@ -61,6 +62,23 @@ function ScrambleMatcher({ wcifEvents, competitionId, initialScrambleFiles }) {
     mutationFn: () => submitMatchedScrambles(competitionId, matchState),
   });
 
+  const roundIds = useMemo(() => wcifEvents.flatMap((event) => event.rounds)
+    .map((r) => r.id), [wcifEvents]);
+
+  const missingIds = useMemo(() => {
+    if (!matchState) return [];
+    return _.difference(roundIds, Object.keys(matchState));
+  }, [matchState, roundIds]);
+
+  const missingScrambleIds = useMemo(() => {
+    if (!matchState) return [];
+    return roundIds.filter((roundId) => {
+      const matchedRound = matchState[roundId];
+      const wcifRound = wcifEvents.flatMap((event) => event.rounds).find((r) => r.id === roundId);
+      return matchedRound.length < wcifRound.scrambleSetCount;
+    });
+  }, [matchState, roundIds, wcifEvents]);
+
   return (
     <>
       <Message info>
```

Comment:
In terms of UX, I believe this whole info message can move down into the upload panel component, and be included there as a simple `Header.Subheader` in a (slightly) condensed form.

```
Scrambles are assigned automatically upon upload. If there is a discrepancy
between the number of scramble sets in the JSON file and the number of groups
in the round you can do manual adjustments below.
```